speakrune/gen/speakrune.hoon

```diff
@@ -0,0 +1,51 @@
+|=  raw=tape
+^-  tape
+=<  ?:((valid raw) (turn raw convert) "input contains non-rune characters")
+|%
+++  valid
+  |=  x=tape
+  ^-  ?
+  =/  chars  (sy x)
+  (~(all in chars) |=(c=@t (~(has by table) c)))
```

Comment:
Since your gate here just calls another gate with the same argument, you can just pass that other gate in instead! (:

```
(~(all in chars) ~(has by table))
```

Reply:
Cool, didn't realize I could do that here.